### PR TITLE
Wallet Server blog post

### PR DESCRIPTION
**Pull request by dgca**
### What changed?

- Adds announcement for the wallet server project

---

<sub>**Reminder:** If content (docs, blogs, pages) is moved or renamed, please ensure there are no broken links.</sub>

### CODE REVIEW

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+title: Introducing the Iron Fish Wallet Server
+description: 
+image: /images/blog/introducing-wallet-server/hero-img.png
+tags: [ironfish]
+---
+
+Today we're excited to announce the initial release of the Iron Fish Wallet Server. This project aims to enable the building and running of a fully private light client without the need to run a local Iron Fish node. In this blog post, we'll provide an overview of the Wallet Server, its key features, and talk about how you can help us improve this project.
+
+## TL;DR:
+
```
Comment (jowparks) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:12]: - A rate limited version of this server is running at walletserver.ironfish.network for testing

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+author: iflabs
+title: Introducing the Iron Fish Wallet Server
+description: 
+image: /images/blog/introducing-wallet-server/hero-img.png
+tags: [ironfish]
+---
+
+Today we're excited to announce the initial release of the Iron Fish Wallet Server. This project aims to enable the building and running of a fully private light client without the need to run a local Iron Fish node. In this blog post, we'll provide an overview of the Wallet Server, its key features, and talk about how you can help us improve this project.
```
Comment (jowparks) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:9]: Might be useful to define what a light client is here for the lay man.

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## TL;DR:
+
+- The Iron Fish Wallet Server is a gRPC server that allows light clients to interact with the Iron Fish network without having to run a full node.
+- The Wallet Server is open source and [available on GitHub](https://github.com/iron-fish/wallet-server).
+- If you're interested in building a light client, talk to us over on Discord on the [#wallet-server](https://discord.com/channels/771503434028941353/1153438621534277785) channel!
+
+## Why do we need a Wallet Server?
+
+Iron Fish is a privacy-first cryptocurrency that uses zero-knowledge proofs to provide fully private transactions. This means that the Iron Fish network does not store any information about the sender, receiver, or amount of a transaction. This is great for privacy, but it also means that you need a full node to interact with the network. This presents a problem for those who want to use Iron Fish from a mobile device, for instance.
```
Comment (jowparks) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:19]:
```suggestion
Iron Fish is a privacy-first cryptocurrency that uses zero-knowledge proofs to provide fully private transactions. This means that the Iron Fish network does not publicly display any information about the sender, receiver, or amount of a transaction. The only person who has access to this data is the person who has the private keys for the account of interest. This is very different than other blockchains like ethereum and bitcoin where anyone can see this data for any account. With the current architecture, it also means that you need to run a full node to privately interact with the network. This makes it impossible for those who want to use Iron Fish accounts on a mobile device, browser, or any low power machine.
```

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+The Wallet Server solves this problem by letting light clients interact with an Iron Fish node without having to run a full node themselves. The Wallet Server is beneficial for both Iron Fish nodes, and light clients. For nodes, it reduces the number of requests they have to handle, and for light clients, it sends them only the information they need to interact with the network.
```
Comment (jowparks) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:21]:
```suggestion
The Wallet Server solves this problem by acting as a proxy between the light client and the full node. The Wallet Server is beneficial for both Iron Fish nodes, and light clients. For nodes, it reduces the number of incoming requests, and light clients will retrieve all the information necessary to construct their account information without having to run a node.
```

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+## How does the Wallet Server work?
+
+The Wallet Server is a gRPC server that talks to one or more Iron Fish nodes. It gets blocks from the Iron Fish nodes, caches them, and provides endpoints to light clients so that they can retrieve the cached blocks.
+
+Light clients can request the latest block, a specific block by hash or sequence, or a range of blocks. The Wallet Server also provides an endpoint for light clients to submit transactions to the network.
```
Comment (jowparks) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:27]:
```suggestion
Light clients can request the latest block, a specific block by hash or sequence, or a range of blocks. The Wallet Server also provides an endpoint for light clients to submit transactions to the network. Note that the transactions received and sent via the wallet server are still completely private.
```

**Review by jowparks — APPROVED** [review]
Generally looks good, I would like to see an image. We should get together with Skylar to see if he understands what this page is saying and if he can come up with some clever, cartoony graphical representation, with fish of course

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+## TL;DR:
+
+- The Iron Fish Wallet Server is a gRPC server that allows light clients to interact with the Iron Fish network without having to run a full node.
+- The Wallet Server is open source and [available on GitHub](https://github.com/iron-fish/wallet-server).
+- If you're interested in building a light client, talk to us over on Discord on the [#wallet-server](https://discord.com/channels/771503434028941353/1153438621534277785) channel!
+
+## Why do we need a Wallet Server?
+
+Iron Fish is a privacy-first cryptocurrency that uses zero-knowledge proofs to provide fully private transactions. This means that the Iron Fish network does not publicly display any information about the sender, receiver, or amount of a transaction. The only person who has access to this data is the person who has the private keys for the account of interest. This is very different than other blockchains like ethereum and bitcoin where anyone can see this data for any account. With the current architecture, it also means that you need to run a full node to privately interact with the network. This makes it difficult to use Iron Fish on a mobile device, browser, or any low power machine.
```
Comment (dguenther) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:19]:
```suggestion
Iron Fish is a privacy-first cryptocurrency that uses zero-knowledge proofs to provide fully private transactions. This means that the only the sender and receiver of a transaction can view the sender, receiver, amount, and asset of a transaction. This is different from other blockchains like Ethereum and Bitcoin, where transaction data is visible to the public. Iron Fish currently requires running a full node to decrypt your transactions, which is possible to do on most computers. However, full nodes have additional computing, bandwidth, and storage responsibilities, like verifying blocks, propagating transactions to the network, and storing the blockchain, that may be inefficient to run on mobile devices or in a browser extension.
```

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+The Wallet Server solves this problem by acting as a proxy between the light client and the full node. The Wallet Server is beneficial for both Iron Fish nodes, and light clients. For nodes, it reduces the number of incoming requests, and light clients will retrieve all the information necessary to construct their account information without having to run a node.
```
Comment (dguenther) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx]:
```suggestion
The Wallet Server reduces the requirements necessary to determine your balance and send transactions by acting as a proxy between light clients and a full node. Light clients are able to download transactions from the Wallet Server, decrypt them to construct their balance, and send transactions without the additional requirements of running a full node.```

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+## How does the Wallet Server work?
+
+The Wallet Server is a gRPC server that talks to one or more Iron Fish nodes. It gets blocks from the Iron Fish nodes, caches them, and provides endpoints to light clients so that they can retrieve the cached blocks.
```
Comment (dguenther) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:28]:
```suggestion
The Wallet Server is a gRPC server that gets blocks from one or more Iron Fish full nodes, strips out data unnecessary for sending transactions and calculating balances, caches them, and provides endpoints that allow light clients to get the cached blocks.
```

File: content/blog/2023-09-19-iron-fish-wallet-server.mdx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+In order to use Iron Fish without running a full node, we need to be able to build a "light client". A light client is a client that does not run a full node, instead it relies on other nodes to provide the information it needs to construct its account information.
+
+This is where the Wallet Server comes in. The Wallet Server acts as a proxy between the light client and the full node, getting block data from the node, caching it, and providing it to light clients. The Wallet Server is beneficial to Iron Fish nodes as well, as it reduces the number of incoming requests they need to process.
+
+## How does the Wallet Server work?
+
+The Wallet Server is a gRPC server that talks to one or more Iron Fish nodes. It gets blocks from the Iron Fish nodes, caches them, and provides endpoints to light clients so that they can retrieve the cached blocks.
+
+Light clients can request the latest block, a specific block by hash or sequence, or a range of blocks. The Wallet Server also provides an endpoint for light clients to submit transactions to the network. Note that the transactions received and sent via the wallet server are still completely private.
```
Comment (dguenther) [inline content/blog/2023-09-19-iron-fish-wallet-server.mdx:30]:
```suggestion
Light clients can request the latest block, a specific block by hash or sequence, or a range of blocks. The Wallet Server also provides an endpoint for light clients to submit transactions to the network. Transactions sent and received via the Wallet Server are still private, as they're encrypted in the same way as when running a full node. The Wallet Server can see which blocks a client is requesting and the encrypted transactions that a client submits, but it cannot see the sender, receiver, amount, or asset of any transactions. Also, light clients do not transfer their account keys to the Wallet Server. 
```

**Pull request merged**